app/models/registration.rb

```diff
@@ -336,6 +338,14 @@ def self.newcomer_month_eligible_competitors_count
     end
   end
 
+  validate :does_not_exceed_competitor_limit
+  private def does_not_exceed_competitor_limit
+    return unless competition&.competitor_limit.present?
+    return unless competing_status == Registrations::Helper::STATUS_ACCEPTED
+    errors.add(:competitor_limit, I18n.t('registrations.errors.competitor_limit_reached')) if
+      competition.registrations.competing_status_accepted.count >= competition.competitor_limit
+  end
```

Comment:
I have an eery feeling that I asked this in other PRs, but... Shouldn't we already be checking this right now (ie before merging your PR, in the status quo production code?)

Reply:
We are already checking this in the registration_checker - what you asked about was a refactoring of where that check happened, which made it appear like it was newly added.

In this case, auto-accepts bypass the registration checker, and so we need to have a registration-level validation to ensure that the competitor limit is respected.

Reply:
- [ ] add a comment that this is a redundant check with the registration_checker, and we want to move the check out of the registration checker when possible because of Fin's PR beiongmerged